### ISSUE TITLE
Unable to start multipass instance in MacOS

### ISSUE THREAD

**Issue by ashok-mariyala** [canonical/multipass#3186]
I have installed multipass using brew in my MacOS. While starting the multipass instance my macbook is automatically restarting. 

Here is the logs

```
tail -500f multipassd.log 
[2023-08-01T12:07:07.453] [warning] [Qt] Empty filename passed to function
[2023-08-01T12:07:07.695] [warning] [Qt] Execution of PAC script at "http%3A%2F%2F127.0.0.1%3A9000%2Flocalproxy-e3b0c442.pac" failed: The operation couldn’t be completed. (NSURLErrorDomain error -1002.)
[2023-08-01T12:07:07.717] [warning] [Qt] Execution of PAC script at "http%3A%2F%2F127.0.0.1%3A9000%2Flocalproxy-e3b0c442.pac" failed: The operation couldn’t be completed. (NSURLErrorDomain error -1002.)
[2023-08-01T12:07:07.742] [warning] [Qt] Execution of PAC script at "http%3A%2F%2F127.0.0.1%3A9000%2Flocalproxy-e3b0c442.pac" failed: The operation couldn’t be completed. (NSURLErrorDomain error -1002.)
[2023-08-01T12:07:09.140] [debug] [update] Latest Multipass release available is version 1.12.2
[2023-08-01T12:07:09.296] [warning] [Qt] Execution of PAC script at "http%3A%2F%2F127.0.0.1%3A9000%2Flocalproxy-e3b0c442.pac" failed: The operation couldn’t be completed. (NSURLErrorDomain error -1002.)
[2023-08-01T12:07:11.290] [warning] [Qt] Execution of PAC script at "http%3A%2F%2F127.0.0.1%3A9000%2Flocalproxy-e3b0c442.pac" failed: The operation couldn’t be completed. (NSURLErrorDomain error -1002.)
[2023-08-01T12:07:11.981] [warning] [Qt] Execution of PAC script at "http%3A%2F%2F127.0.0.1%3A9000%2Flocalproxy-e3b0c442.pac" failed: The operation couldn’t be completed. (NSURLErrorDomain error -1002.)
[2023-08-01T12:07:12.182] [warning] [Qt] Execution of PAC script at "http%3A%2F%2F127.0.0.1%3A9000%2Flocalproxy-e3b0c442.pac" failed: The operation couldn’t be completed. (NSURLErrorDomain error -1002.)
[2023-08-01T12:07:12.190] [warning] [Qt] Execution of PAC script at "http%3A%2F%2F127.0.0.1%3A9000%2Flocalproxy-e3b0c442.pac" failed: The operation couldn’t be completed. (NSURLErrorDomain error -1002.)
[2023-08-01T12:07:12.191] [info] [VMImageHost] Did not find any supported products in "appliance"
[2023-08-01T12:07:12.195] [debug] [blueprint provider] Loading "anbox-cloud-appliance" v1
[2023-08-01T12:07:12.196] [debug] [blueprint provider] Loading "charm-dev" v1
[2023-08-01T12:07:12.197] [debug] [blueprint provider] Loading "docker" v1
[2023-08-01T12:07:12.197] [debug] [blueprint provider] Loading "jellyfin" v1
[2023-08-01T12:07:12.198] [debug] [blueprint provider] Loading "minikube" v1
[2023-08-01T12:07:12.198] [debug] [blueprint provider] Loading "ros-noetic" v1
[2023-08-01T12:07:12.198] [debug] [blueprint provider] Loading "ros2-humble" v1
[2023-08-01T12:07:12.398] [info] [rpc] gRPC listening on unix:/var/run/multipass_socket
[2023-08-01T12:07:12.481] [debug] [qemu-img] [884] started: qemu-img snapshot -l /var/root/Library/Application Support/multipassd/qemu/vault/instances/primary/ubuntu-22.04-server-cloudimg-arm64.img
[2023-08-01T12:07:12.925] [info] [daemon] Starting Multipass 1.12.2+mac
[2023-08-01T12:07:12.925] [info] [daemon] Daemon arguments: /Library/Application Support/com.canonical.multipass/bin/multipassd --verbosity debug
```
Here is the multipass version
```
multipass version
multipass   1.12.2+mac
multipassd  1.12.2+mac
```

**Comment by georgeliao** [comment]
Hi @ashok-mariyala , thanks for the report, can you share more details regarding to which step causes the macOS restart? Is it `multipass launch <instance_name>` or `multipass start <instance_name>`? And also share other info based on the format of the report, like macOS version and local.driver (`multipass get local.driver`).

**Comment by ashok-mariyala** [comment]
@georgeliao Thank you for your quick response. I am facing issue in both cases multipass launch <instance_name> and multipass start <instance_name> 
**OS Informarion:** `MacOS Ventura version 13.5`

**Local Driver**
```
multipass get local.driver
qemu
```

Please let me know if you need further information. Thanks in advance.

**Comment by cemakpolat** [comment]
https://github.com/canonical/multipass/issues/3182 ticket that I wrote is nearly the same.

**Comment by townsend2010** [comment]
Hey @cemakpolat!

You mean in #3182, your machine reboots as well when launching an instance?

**Comment by georgeliao** [comment]
Hi @ashok-mariyala 

It looks like the system update and the multipass version has something to do with that. Can you try older version (not from brew, but from the downloaded darwin package (https://github.com/canonical/multipass/releases)) of multipass and see it the same problem persists? Besides that, can you also provide a system log before the reboot so we can understand better about what is going on?

**Comment by cemakpolat** [comment]
@townsend2010 no, this is not the case in my ticket. The core issue is in my ticket, after fetching the image, the starting process fails. I cannot deliver too much details, since I believe it happens nothing :)

**Comment by cemakpolat** [comment]
@georgeliao I tried the older version as well, the same issue occurred.

**Comment by alex-bluesman** [comment]
Hi all,

I could only guess, but spent today the whole day debugging similar issue on MacOS 13.5.

The behavior is pretty similar:
1. I start instance
2. I see in process list and Multipass logs that QEMU is started
3. After few minutes I got timeout error

What I finally did:
1. Boot the QEMU manually using the command from 'ps ax | grep multi' with console mounting to terminal:

'''
/Library/Application\ Support/com.canonical.multipass/bin/qemu-system-aarch64 -mm
achine virt,gic-version=3 -accel hvf -drive file=/Library/Application\ Support/cc
om.canonical.multipass/bin/../Resources/qemu/edk2-aarch64-code.fd,if=pflash,formm
at=raw,readonly=on -cpu host -nic vmnet-shared,model=virtio-net-pci,mac=52:54:000
:78:ab:fc -device virtio-scsi-pci,id=scsi0 -drive file=/var/root/Library/Applicaa
tion\ Support/multipassd/qemu/vault/instances/phlegmatic-agama/ubuntu-22.04-servv
er-cloudimg-arm64.img,if=none,format=qcow2,discard=unmap,id=hda -device scsi-hd,,
drive=hda,bus=scsi0.0 -smp 4 -m 8192M -nographic -cdrom /var/root/Library/Applicc
ation\ Support/multipassd/qemu/vault/instances/phlegmatic-agama/cloud-init-confii
g.iso
'''

2. I noticed that there is no IPv4 address set for the ethernet adapter, only IPv6
3. If I set IPv4 address manually and DO NOT stop the QEMU, then I could connect to this instance using 'Multipass shell'

So my rough guess is that after recent MacOS update there is an issue with firewall and instances doesn't receive IP address.
